.gitlab-ci.yml

```diff
@@ -156,21 +183,34 @@ publish:s3:
     - export AWS_SECRET_ACCESS_KEY=$TMP_STORAGE_AWS_SECRET_ACCESS_KEY
     - apt update && apt install -yyq awscli
     # Fetch artifacts from temporary S3 bucket
-    - aws s3 cp s3://mender-gitlab-tmp-storage/$CI_PROJECT_NAME/$CI_PIPELINE_ID/deploy.tar.gz deploy.tar.gz
-    - tar xzf deploy.tar.gz
+    - aws s3 cp s3://mender-gitlab-tmp-storage/$CI_PROJECT_NAME/$CI_PIPELINE_ID/raspberrypi3.tar.gz raspberrypi3.tar.gz
+    - tar xzf raspberrypi3.tar.gz
   script:
     # Prepare high privilege S3 keys (the TMP_STORAGE keys are for the tmp storage only)
     - export AWS_ACCESS_KEY_ID=$PUBLISH_AWS_ACCESS_KEY_ID
     - export AWS_SECRET_ACCESS_KEY=$PUBLISH_AWS_SECRET_ACCESS_KEY
-    - echo "Publishing ${RASPBIAN_NAME}-mender-${MENDER_CLIENT_VERSION}.img.xz version to S3"
-    - aws s3 cp deploy/${RASPBIAN_NAME}-raspberrypi3-mender.img.xz
-        s3://$S3_BUCKET_NAME/${RASPBIAN_NAME}/arm/${RASPBIAN_NAME}-mender-${MENDER_CLIENT_VERSION}.img.xz
+
+    - echo "Publishing ${RASPBIAN_NAME}-raspberrypi4-mender-${MENDER_CLIENT_VERSION}.img.xz version to S3"
```

Comment:
`raspberrypi3` here

Reply:
I guess the best would be to have the string `${RASPBIAN_NAME}-raspberrypi4-mender-${MENDER_CLIENT_VERSION}.img.xz` into an env variable, so that we are certain we echo and cp the same thing

Reply:
Good point. I took the opportunity to make this even better, and turned both sections in the publish stage into loops. You can actually make loops across several list entries in the yaml. I think that wasn't possible in the past, I can swear I tried it a while ago. But it works now, so take a look!